### ISSUE TITLE
Flood fill example has a bug.

### ISSUE THREAD

**Issue by whoiscc** [photonstorm/phaser3-examples#22]
The game is [here](https://labs.phaser.io/view.html?src=src\games\flood\flood-fill.js). The bug occurs randomly, and more frequently if the screen recorder is not enabled.
![2018-03-08_22-47-30-1-1](https://user-images.githubusercontent.com/35130945/37157547-1a8c292c-2324-11e8-996d-82a8260693fa.gif)

**Comment by Rybar** [comment]
I've noticed this bug as well, haven't been able to pin down the exact conditions that cause it, other than rapidly entering and exiting the mouseover state for each icon.

**Comment by Matthew-Herman** [comment]
Copying this from a duplicate issue:
I have been looking at this bug and it seems to be an issue with stopping the tween that plays once a user mouses over a monster icon.

On line 364 we have this function

    onIconOut: function (pointer, gameObject)
    {
        console.log(this.monsterTween.targets[0].y);

        //this line causes monster.y to become 0 when cursor moves erratically 
        this.monsterTween.stop(0);

        console.log(this.monsterTween.targets[0].y);

        this.cursorTween = this.tweens.add({
            targets: this.cursor,
            alpha: 0,
            duration: 300
        });

        this.arrow.setFrame('arrow-white');
    },
It is certainly the behvaior of this.monsterTween.stop(0);, which causes the tween to stop but sometimes erroneously sets the y position of the monster icon to 0. I'm struggling to find out why this is because I don't know how stopping Tweens is implemented.

**Comment by gabegordon** [comment]
One potential fix for this issue is resetting the tween target to the original icon y position if it becomes set to 0.
For example adding the following anywhere in onIconOver or onIconOut:
```
if (this.hasOwnProperty('monsterTween') &&  this.monsterTween.targets[0].y === 0) {
    this.monsterTween.targets[0].y = icon.y;
}
```

**Comment by photonstorm** [comment]
Yes, that would work. I would rather find out why seeking a Tween sometimes, in some circumstances, causes the tween property to reset to zero though. It feels like a more useful bug to resolve :)
